x-pack/plugin/esql/src/main/java/org/elasticsearch/xpack/esql/inference/rerank/RerankOperator.java

```diff
@@ -0,0 +1,128 @@
+/*
+ * Copyright Elasticsearch B.V. and/or licensed to Elasticsearch B.V. under one
+ * or more contributor license agreements. Licensed under the Elastic License
+ * 2.0; you may not use this file except in compliance with the Elastic License
+ * 2.0.
+ */
+
+package org.elasticsearch.xpack.esql.inference.rerank;
+
+import org.elasticsearch.compute.data.Block;
+import org.elasticsearch.compute.data.DoubleBlock;
+import org.elasticsearch.compute.data.Page;
+import org.elasticsearch.compute.operator.DriverContext;
+import org.elasticsearch.compute.operator.EvalOperator.ExpressionEvaluator;
+import org.elasticsearch.compute.operator.Operator;
+import org.elasticsearch.core.Releasables;
+import org.elasticsearch.threadpool.ThreadPool;
+import org.elasticsearch.xpack.esql.inference.InferenceOperator;
+import org.elasticsearch.xpack.esql.inference.InferenceRunner;
+import org.elasticsearch.xpack.esql.inference.bulk.BulkInferenceExecutionConfig;
+
+import java.util.stream.IntStream;
+
+/**
+ * {@link RerankOperator} is an inference operator that compute scores for rows using a reranking model.
+ */
+public class RerankOperator extends InferenceOperator {
+
+    // Default number of rows to include per inference request
+    private static final int DEFAULT_BATCH_SIZE = 20;
```

Comment:
do we know if the rerank API has any limitations when sending large docs?
we might want to control the batch size not just by the number of rows, but on the total input size. 
okay with me to follow up on this separately